api/src/main/java/org/openmrs/api/ConceptService.java

```diff
@@ -1922,5 +1922,57 @@ public Drug getDrugByMapping(String code, ConceptSource conceptSource,
 	 */
 	@Authorized(PrivilegeConstants.GET_CONCEPTS)
 	boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType);
-	
+	/**
+	 * Get Concept by id or name convenience method
+	 * 
+	 * @param idOrMap
+	 * @return the found Concept
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_CONCEPTS)
+	public static Concept getConceptInfo(String idOrMap) throws APIException {
+
+		Concept cpt = null;
+
+		if (idOrMap != null) {
```

Comment:
I would do the opposite check and throw the exception right away. In that case, less indentation makes code more easy for your eyes. 👀 something like below(no need to write `else` case):
```
if(idOrName == null) {
  throw new APIException("null isn`t alloed as acceptable input");
}
```

Reply:
@JudeNiroshan
 ``if(idOrName == null) {
  throw new APIException("null isn`t alloed as acceptable input");
}``
this is way more easier to read, thanks

Reply:
https://gph.is/g/ap0YXzx

Reply:
There is a small typo -> `null isn't allowed as...`